### PR TITLE
feat: upgrade the opentelemetry api to 1.0.0

### LINKED ISSUE #1330: Upgrade opentelemetry api to the 1.0.0 version
Thanks for stopping by to let us know something could be better!

1) Is this a client library issue or a product issue?
This is a client library issue

2) Did someone already solve this?
Yes, I have resolved it in the past :)

3) Do you have a support contract?
No

#### Environment details

  - OS: macOS 11.4.0
  - Node.js version: 14.17.0
  - npm version: 6.14.11
  - `@google-cloud/pubsub` version: 2.14.0

#### Steps to reproduce

None, this is a dependency upgrade to ensure the opentelemetry integration is using the 1.0.0 version which introduces semver and backward-compatibility requirements.

Making sure to follow these steps will guarantee the quickest resolution possible.

Thanks!

### PR DESCRIPTION
**Pull request by weyert**
Updates the `@opentelemetry/api`-package to 1.0.0 and use the associated
SDK version which supports this API version that is: 0.22.0

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [X] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-pubsub/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [X] Ensure the tests and linter pass
- [X] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #1330 🦕

### CODE REVIEW

**Comment by tapico-weyert** [comment]
@googlebot I consent

File: samples/package.json [hunk 1]
```diff
@@ -25,9 +25,9 @@
     "updatejs": "cp build/*.js ."
   },
   "dependencies": {
-    "@google-cloud/pubsub": "^2.14.0",
```
Comment (feywind) [inline samples/package.json:28]: I think this got reverted somewhere... maybe just needs a merge from the `master` branch?

Comment (weyert) [reply samples/package.json:28]: Resolved it :)
One day I will remember which option to pick (Incoming vs Current changes) in Visual Studio Code :)

Comment (feywind) [reply samples/package.json:28]: I get confused by that terminology a lot. It seemed like it was more straightforward in the Perforce world, somehow.

**Review by feywind — APPROVED** [review]

**Pull request merged**